src/ui/components/FormOverlay/index.js

```diff
@@ -0,0 +1,157 @@
+/* @flow */
+/* eslint-disable react/sort-comp */
+import makeClassName from 'classnames';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import { closeFormOverlay } from 'core/reducers/formOverlay';
+import translate from 'core/i18n/translate';
+import Button from 'ui/components/Button';
+import Icon from 'ui/components/Icon';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+import type { FormOverlayState } from 'core/reducers/formOverlay';
+
+import './styles.scss';
+
+type Props = {|
+  children?: React.Node,
+  className?: string,
+  dispatch: DispatchFunc,
+  i18n: I18nType,
+  id: string,
+  isOpen: boolean,
+  isSubmitting: boolean,
+  onCancel?: Function,
+  onSubmit?: Function,
+  submitText?: string,
+  submittingText?: string,
+  title: string,
+|};
+
+export class FormOverlayBase extends React.Component<Props> {
+  static defaultProps = {
+    isOpen: true,
+  };
+
+  closeOverlay(event: SyntheticEvent<any>) {
+    const { id, dispatch } = this.props;
+    event.preventDefault();
```

Comment:
Usually we just need `preventDefault()`. Is there a reason both are needed here? I believe it, but just would be cool to know why 😄

Reply:
Since we have a lot of handlers all over the app for the same events (like clicks), it can be tricky to predict which handlers will fire as a result of event bubbling. As a precaution, I think we should always add `event.stopPropagation()` to stop bubbling until we know we need bubbling. More info: https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation